docs/getting-started/auto-suggest.md

```diff
@@ -0,0 +1,212 @@
+1. TOC
+{:toc}
+</details>
+
+---
+
+## Introduction
+
+> Introduced in Tasks 1.9.0.
+
+The [Priorities]({{ site.baseurl }}{% link getting-started/priority.md %}), [Dates]({{ site.baseurl }}{% link getting-started/dates.md %}) and [Recurring Tasks]({{ site.baseurl }}{% link getting-started/recurring-tasks.md %}) pages show various emojis and special phrases that the Tasks plugin recognises, when searching for tasks.
+
+If you prefer to type your tasks, instead of using a dialog, there is now an intelligent auto-suggest completion mechanism that does a
+lot of the typing of emojis and dates for you.
+
+It is particularly powerful when creating and editing tasks on mobile phones.
+
+### Video Demo
+
+It is perhaps best understood by watching a [video of it in action](https://user-images.githubusercontent.com/10722656/175102574-78b0f851-cc48-4255-a40e-d3036bec5bb6.gif).
+
+### Walk Through
+
+Here is a more detailed walk through of the creation of a new task, which can be done entirely using the keyboard if you wish.
+
+1. As you are type a task, the auto-complete menu pops up to show some common options:
+
+    ![auto-suggest-menu-initial-menu](https://github.com/obsidian-tasks-group/obsidian-tasks/raw/docs-document-typing-tasks/resources/screenshots/auto-suggest-menu-initial-menu.png)
+
+2. You can keep typing, or select one of the menu items in a variety of ways:
+
+    - mouse-click on menu item
+    - use the up/down keyboard keys and then type `Return` or `Enter`
+    - type a few more characters, to make the menu items more specific. For example, type `pri` to show all the options for setting the task's priority.
+
+3. Here we selected the 'high priority' item, and so now the menu is updated to show the next most likely items you might want to add. We are going to select 'recurring (repeat)' from this menu:
+
+    ![auto-suggest-menu-after-priority](https://github.com/obsidian-tasks-group/obsidian-tasks/raw/docs-document-typing-tasks/resources/screenshots/auto-suggest-menu-after-priority.png)
+
+4. Now the repeat emoji has been added, the menu offers a few common options for recurrences:
+
+    ![auto-suggest-menu-after-repeat-emoji](https://github.com/obsidian-tasks-group/obsidian-tasks/raw/docs-document-typing-tasks/resources/screenshots/auto-suggest-menu-after-repeat-emoji.png)
+
+5. We chose `every week`, which was added to our task, and now the menu shows the remaining possible emojis:
+
+    ![auto-suggest-menu-after-repeat-text](https://github.com/obsidian-tasks-group/obsidian-tasks/raw/docs-document-typing-tasks/resources/screenshots/auto-suggest-menu-after-repeat-text.png)
+
+6. We selected the 'due date', and so now the menu offers a selection of commonly-used dates, calculated based on the current date:
+
+    ![auto-suggest-menu-after-due-emoji](https://github.com/obsidian-tasks-group/obsidian-tasks/raw/docs-document-typing-tasks/resources/screenshots/auto-suggest-menu-after-due-emoji.png)
+
+7. Now our task is complete:
```

Comment:
Maybe you also want to add that in cases where the menu starts with the return sign ⏎ (which is when only "generic", i.e. non-matches, suggestions are shown), it is given as a default option to enter a new line instead of choosing a suggestion.

Reply:
It will not, because the Return sign is shown only when there is no concrete match (assuming the user has no benefit in choosing the 1st suggestion).

Reply:
I've attempted to summarise this in 3c5e8122e998fb29d9810df8e0ccddfd648a9ab6 - but it's the one area that I feel I don't really understand fully, so it may not be as clear as it could be.